hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/impl/OperationThread.java

```diff
@@ -132,20 +132,27 @@ public final void executeRun() {
 
     private void process(Object task) {
         try {
+            SwCounter counter;
```

Comment:
You are right that it conflates the return value. However your proposal will lead to a lot of code duplication and opportunity for errors. I think my solution is elegant, even though a bit confusing. I wish we had tuples :) however they do bring some issues as well

Reply:
I will put in a comment to un-confuse

Reply:
Is there an example how much code duplication it actually introduces? I don't think it'll be too much and I must agree this approach is suprising to say the least. It also requires you to do code like this, which might not be necessary if you were passing around boolean:
`return currentRunner.run(operation) ? null : completedOperationCount;`

Reply:
It saves at least 12 redundant lines of code in hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/impl/OperationThread.java, you can see the changes right in this PR.

The issue here isn't only the therwise-would-be 12 lines of code, the alternative would require creating an if() statement for every one of those counter instances to check if the operation is available, and only then increment the counter, leading to another 20+ line of code duplication, which is very error-prone and would require another 50+ lines of unit tests.

In reality it may not be that complex, but just thinking about it makes this solution much more elegant

Reply:
Can you point to the commit where the alternate solution is present? I tried going through the git history for this class but could not find it. The issue being that the current codebase definitely has issues with clarity and I'm unsure if I can assess how worse the alternate solution is.

Reply:
Unfortunately when I did the alternative solution it was pretty bad and I didn't even commit it.
I think the comment makes the current solution pretty  clear on what it does.
I can roll this whole thing back and put back the "bad" solution but I need some assurances that the work will not be in vain. I need this PR done in 4.1 and if that's what it'll take I can do that (reluctantly)

Reply:
Ok, this is taking too much time and energy.
I'll jut revert this change, put in all the duplicate code that was here before
so we can just move on from this.